### DISCUSSION TITLE
UI and Feature Suggestions

### DISCUSSION THREAD

**Discussion by ManedCalico** [mcarlucci/decky-storage-cleaner#3] [Ideas]
Hi there!

First, I want to say that I'm so glad you've made this plug-in! It's incredibly handy to be able to clean up compat/shadercache without using a script or needing to switch to Desktop. It's also made it a lot easier to help less tech-savvy friends clean up their other storage space.

That said, I have a few ui and feature suggestions for you, if that's alright. I'm not sure how easy it would be to implement these things, but I figured it wouldn't hurt to share some ideas. I feel these would help the plugin be more user friendly if they're possible.

1. It would be great if there were either different sections or a view filter to separate currently installed vs uninstalled games. For the most part, I don't think there's much need for the average person to clean up files for installed games, and the list of installed games can get very long on larger Decks / sd cards

2. It would also be helpful if there were separate sections for compat and shadercache, instead of both being in one list together

3. I think the best way to handle the two above suggestions would be to utilize the full Deck screen instead of having everything present in the QAM

4. At the moment, the two clear all buttons are underneath their relative sections. I worry someone might accidentally hit them instead of clear selected without realizing. I think it would be best to have them grouped together somewhere separately. If you are able to create a full screen browser for selecting individual games, maybe these quick options could stay in the QAM below a "select games" button

4. Adding a "clear all uninstalled" button would also be really helpful

5. I've seen other plugins inject their own game options menu items (SGDB's "change artwork" for example). If the plugin could add an "Uninstall + Clean" option under the Manage options menu to uninstall a game and also delete its shadercache/compat at the same time, that would be absolutely amazing!

Thank you!!!

**Comment by mcarlucci** [comment]
> Hi there!
> 
> First, I want to say that I'm so glad you've made this plug-in! It's incredibly handy to be able to clean up compat/shadercache without using a script or needing to switch to Desktop. It's also made it a lot easier to help less tech-savvy friends clean up their other storage space.
> 
> That said, I have a few ui and feature suggestions for you, if that's alright. I'm not sure how easy it would be to implement these things, but I figured it wouldn't hurt to share some ideas. I feel these would help the plugin be more user friendly if they're possible.
> 
> 1. It would be great if there were either different sections or a view filter to separate currently installed vs uninstalled games. For the most part, I don't think there's much need for the average person to clean up files for installed games, and the list of installed games can get very long on larger Decks / sd cards
> 2. It would also be helpful if there were separate sections for compat and shadercache, instead of both being in one list together
> 3. I think the best way to handle the two above suggestions would be to utilize the full Deck screen instead of having everything present in the QAM
> 4. At the moment, the two clear all buttons are underneath their relative sections. I worry someone might accidentally hit them instead of clear selected without realizing. I think it would be best to have them grouped together somewhere separately. If you are able to create a full screen browser for selecting individual games, maybe these quick options could stay in the QAM below a "select games" button
> 5. Adding a "clear all uninstalled" button would also be really helpful
> 6. I've seen other plugins inject their own game options menu items (SGDB's "change artwork" for example). If the plugin could add an "Uninstall + Clean" option under the Manage options menu to uninstall a game and also delete its shadercache/compat at the same time, that would be absolutely amazing!
> 
> Thank you!!!

Thanks for the feedback, suggestions and ideas! 

1. This is a great idea. I should have what I need from the Steam JS SDK to support this. 

2. This has been a major consideration of mine lately, so thanks for validating it. I intend to move the main content to a page route for more digital real estate, and instead use the Decky slideout area for more quick actions and from there be able to perform more "advanced" actions (in the full page route).

3.  [addressed in 2]

5. Yeah, this is mainly why I added the confirm modals, because of the destructive nature of the clearing actions. That and it is good UX to have users confirm destructive actions in general, as a principle. To your point, I believe I can solve this in the full page route. 

6. Good idea. Think this would be good to have as a Quick Action button in the slideout area?

7. Oooooohhhh this is a great idea! Love it!

I have some work to do. Thanks again! 😄

**Reply by ManedCalico** [reply]
Thank you so much for liking these suggestions!

1. Awesome! I'm glad this might be an easy feature to add!

2. I'm happy to have validated your idea! I definitely think using a page route for the more granular, advanced features while keeping quick actions in the slide out is a great way to go!

3. [above]

4. Moving to a page would make this a non-issue for sure

5. Absolutely! It would be incredibly handy to have this quickly available in the slide out

6. Thank you!! Definitely a shoot for the moon type idea, but hopefully possible!

Hopefully I haven't given you too much of a headache with all this, but thank you again for being open to looking into these!
